packages/artillery-plugin-publish-metrics/lib/open-telemetry/index.js

```diff
@@ -103,19 +103,32 @@ class OTelReporter {
     userContext.vars['__otlStartTime'] = startTime;
     const span = trace
       .getTracer('artillery-tracer')
-      .startSpan('http_request', { startTime });
-    span.setAttribute('kind', 'client');
+      .startSpan(req.method.toLowerCase(), {
+        startTime,
+        kind: SpanKind.CLIENT
+      });
+
     span.addEvent('http_request_started', startTime);
     userContext.vars['__otlpSpan'] = span;
 
+    events.on('error', (err) => {
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: err.message || err
+      });
+      debug('Span status set as error due to the following error: \n', err);
+      span.end(Date.now);
+      debug('Span finished');
+    });
```

Comment:
Again, by convention, the span status must be set as an error in the case of status codes of over 400 or other errors, so I've implemented that here and on line 154-156.  https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status

This makes errors now appear properly on platforms like Honeycomb 🥳